File: auth-plugin/app/auth/implementors/PlayAuthCache.java
package auth.implementors;

import auth.interfaces.IAuthCache;
import auth.models.AuthenticatedUser;
import com.google.inject.Inject;
import play.cache.CacheApi;
import play.libs.Json;

import java.util.UUID;

public class PlayAuthCache implements IAuthCache {

    static int expiryTimeAmount = 60 * 60 * 24 * 7;

    CacheApi cache;

    @Inject
    public PlayAuthCache(CacheApi playCache) {
        this.cache = playCache;
    }

    @Override
    public boolean isUserInCache(String session) {
        return cache.get(session) != null;
    }

    @Override
    public String storeUserInCache(AuthenticatedUser authenticatedUser) {
        String session = UUID.randomUUID().toString();
        cache.set(session, authenticatedUser.toJson(), expiryTimeAmount);
        return session;
    }

    @Override
    public AuthenticatedUser getUserFromCache(String session) {
        String json = cache.get(session);

        if(json == null)
            return null;

        return AuthenticatedUser.fromJson(json);
    }

    @Override
    public boolean removeUserFromCache(String session) {
        cache.remove(session);
        return true;
    }

    @Override
    public boolean refreshUserAuthenticationSession(String session) {
        String json = cache.get(session);

        if(json == null)
            return false;

        cache.set(session, json, expiryTimeAmount);
        return true;
    }
}

File: auth-plugin/app/auth/implementors/PlayAuthCache.scala
package auth.implementors

import auth.interfaces.IAuthCache
import auth.models.AuthenticatedUser
import javax.inject.Inject
import play.api.Logger
import play.api.cache.CacheApi
import play.libs.Json
import scala.concurrent.duration._
import java.util.UUID

case class PlayAuthCache @Inject()(cache: CacheApi) extends IAuthCache {

  //By having an internal cache, we limit the amount of calls to the Authentication API
  //If this is not an issue then turn this cache off
  val expiryTimeAmount:Int = 2;

  override def isUserInCache(session: String):Boolean = {
    cache.get[String](session) match {
      case Some(_) => true
      case None => false
    }
  }

  override def storeUserInCache(authenticatedUser: AuthenticatedUser):String = {
    val session: String = UUID.randomUUID().toString()
    cache.set(session, authenticatedUser.toJson(), Duration(expiryTimeAmount, SECONDS))
    session
  }

  override def getUserFromCache(session: String):AuthenticatedUser = {
    cache.get[String](session) match {
      case Some(json) => AuthenticatedUser.fromJson(json)
      case None => null
    }
  }

  override def removeUserFromCache(session: String):Boolean = {
    cache.remove(session)
    true
  }

  override def refreshUserAuthenticationSession(session: String):Boolean = {
    cache.get[String](session) match {
      case Some(json) => {
        cache.set(session, json, Duration(expiryTimeAmount, SECONDS))
        true
      }
      case None => false
    }
  }
}
